apps/gitness/src/pages-v2/pull-request/pull-request-compare.tsx

```diff
@@ -313,12 +313,23 @@ export const CreatePullRequest = () => {
     [branchList, tagsList]
   )
 
+  const { setTagList, setBranchList, setRepoId, setSpaceId } = useRepoBranchesStore()
+
+  useEffect(() => {
+    setTagList(tagsList)
+    setBranchList(branchList)
```

Comment:
Can these be moved to the onsuccess callback of the query instead of a useEffect?

Reply:
Ohh yes, I actually checked that since you mentioned it in the previous comment, but it seems the onSuccess callback is deprecated due to performance overheads and will be removed in the next major version of react-query. We'll need to rely on the data object as we are doing now.

Reply:
good point. Interesting read: https://tkdodo.eu/blog/breaking-react-querys-api-on-purpose